gammapy/spectrum/flux_point.py

```diff
@@ -802,13 +805,17 @@ def __init__(
 
         self.datasets = datasets.copy()
         self.e_edges = e_edges
+
+        dataset = self.datasets.datasets[0]
```

Comment:
Yes, that's a good question. I guess this can happen e.g. for joint-likelihood fits, when a "secondary" source is in the FOV of one observation, but not in the other. For the use-case of flux points estimation, I think it's probably save to assume, that the source of interest is part of all datasets and it's probably OK to keep it here.